### DISCUSSION TITLE
Make capture buffer unique

### DISCUSSION THREAD

**Discussion by qingshuizheng** [ahyatt/ekg#56] [Ideas]
Sometimes I need to start a second capture, but keep first capture open, but the two instances would tangle together. 

It is due to `ekg-capture` setting buffer not uniquely, with `(let ((buf (get-buffer-create "*EKG Capture*))) ...)`.

Line changes below would make it feasible. If you think they are okay to take into the package, please add them 😃. 

```elisp
(defun ekg-capture (&optional tags properties subject)
  "Capture a new note, with TAGS and other PROPERTIES.
If SUBJECT is given, force the triple subject to be that value."
  (interactive)
  ;; ↓
  ;; (let ((buf (get-buffer-create "*EKG Capture*")))
  (let* ((subject (or subject (ekg--generate-id)))
         (buf (get-buffer-create (format "*EKG Capture: %s*" subject))))
  ;; ↑
    (set-buffer buf)
    (funcall ekg-capture-default-mode)
    (ekg-capture-mode 1)
    (setq ekg-note
          (ekg-note-create nil ekg-capture-default-mode
                           (seq-uniq (append
                                      tags
                                      (mapcan (lambda (f) (funcall f)) ekg-capture-auto-tag-funcs)))))
    ;; ↓
    ;; (when subject
    ;;  (setf (ekg-note-id ekg-note) subject))
    (setf (ekg-note-id ekg-note) subject)
    ;; ↑
    (setf (ekg-note-properties ekg-note) properties)
    (ekg-edit-display-metadata)
    (goto-char (point-max))
    (mapc (lambda (tag) (run-hook-with-args 'ekg-note-add-tag-hook tag)) (ekg-note-tags ekg-note))
    (pop-to-buffer buf)))
```

Zheng

**Comment by ahyatt** [comment]
Thanks, I've noticed the same problem.  I think uniquify might be another way to solve this without much or any code, let me think about it.

**Comment by qingshuizheng** [comment]
No worry. I’ve hacked it in my personal init. Just mention it here in case you don’t know.

Thanks!

**Reply by ahyatt** [reply]
Sure, but it'd be good is users didn't have to do that, though.

**Reply by ahyatt** [reply]
I've checked in a fix to the develop branch, thank you for suggested this!
